packages/hardhat-core/src/internal/hardhat-network/provider/provider.ts

```diff
@@ -185,11 +188,11 @@ export class EdrProviderWrapper
       initializeVmTraceDecoder(this._vmTraceDecoder, tracingConfig);
     }
 
-    _provider.setOverrideCallback((address: Buffer) => {
-      let overridenResult: Buffer | undefined;
+    _provider.setOverrideCallback((address: Buffer, data: Buffer) => {
+      let overridenResult: ReturnType<AfterMessageCallback>;
 
       for (const afterMessageCallback of this._afterMessageCallbacks) {
```

Comment:
Do we want to allow multiple subscribers when the function explicitly is intended to be mutable?

Reply:
After saying that, I realized that throwing is a bad idea. Smock does set the callback a second time if the network is reset. That's also a good reason to _not_ use an array of callbacks in this case, since every reset would add an unnecessary redundant callback to the list.